ttnn/cpp/ttnn/operations/ccl/reduce_scatter/device/host/reduce_scatter_full_worker_grid.cpp

```diff
@@ -750,13 +752,17 @@ operation::ProgramWithCallbacks reduce_scatter_with_workers(
         input_tensor_n_elems_per_slice / (tt::constants::TILE_WIDTH * tt::constants::TILE_HEIGHT);
 
     TT_ASSERT(input_tensor_num_units_per_tensor_slice > 0);
-    uint32_t max_num_workers = std::min<std::size_t>(8, input_tensor_num_units_per_tensor_slice);
+    uint32_t max_num_workers = std::min<std::size_t>(user_defined_num_workers.value_or(input_tensor_num_units_per_tensor_slice), input_tensor_num_units_per_tensor_slice);
```

Comment:
This behaviour will be different from before? Is this intended?

Reply:
good catch! this should be 
```
    uint32_t max_num_workers = std::min<std::size_t>(user_defined_num_workers.value_or(8), input_tensor_num_units_per_tensor_slice);
```
I will fix that